src/wazuh_db/wdb_parser.c

```diff
@@ -443,7 +470,173 @@ int wdb_parse(char * input, char * output) {
+                mdebug2("Global DB query error near: %s", query);
+                snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", query);
+                result = OS_INVALID;
+            } else {
+                result = wdb_parse_global_update_agent_group(wdb, next, output);
+            }
+        } else if (strcmp(query, "find-group") == 0) {
+            if (!next) {
+                mdebug1("Global DB Invalid DB query syntax.");
+                mdebug2("Global DB query error near: %s", query);
+                snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", query);
+                result = OS_INVALID;
+            } else {
+                result = wdb_parse_global_find_group(wdb, next, output);
+            }
+        } else if (strcmp(query, "insert-agent-group") == 0) {
+            if (!next) {
+                mdebug1("Global DB Invalid DB query syntax.");
+                mdebug2("Global DB query error near: %s", query);
+                snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", query);
+                result = OS_INVALID;
+            } else {
+                result = wdb_parse_global_insert_agent_group(wdb, next, output);
+            }
+        } else if (strcmp(query, "insert-agent-belong") == 0) {
+            if (!next) {
+                mdebug1("Global DB Invalid DB query syntax.");
+                mdebug2("Global DB query error near: %s", query);
+                snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", query);
+                result = OS_INVALID;
+            } else {
+                result = wdb_parse_global_insert_agent_belong(wdb, next, output);
+            }
+        } else if (strcmp(query, "delete-group-belong") == 0) {
+            if (!next) {
+                mdebug1("Global DB Invalid DB query syntax.");
+                mdebug2("Global DB query error near: %s", query);
+                snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", query);
+                result = OS_INVALID;
+            } else {
+                result = wdb_parse_global_delete_group_belong(wdb, next, output);
+            }
+        } else if (strcmp(query, "delete-group") == 0) {
+            if (!next) {
+                mdebug1("Global DB Invalid DB query syntax.");
+                mdebug2("Global DB query error near: %s", query);
+                snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", query);
+                result = OS_INVALID;
+            } else {
+                result = wdb_parse_global_delete_group(wdb, next, output);
+            }
+        } else if (strcmp(query, "select-groups") == 0) {
+            result = wdb_parse_global_select_groups(wdb, output);
+        } else if (strcmp(query, "select-keepalive") == 0) {
+            result = wdb_parse_global_select_agent_keepalive(wdb, next, output);
```

Comment:
It looks there was a merge problem.  wdb_parse_global_select_agent_keepalive() should have the same argument checks as other parsers.
if (!next) {

Reply:
Fixed!